src/swarm-scenario/Swarm/Game/Robot.hs

```diff
@@ -276,6 +277,9 @@ inventoryHash = to (\r -> 17 `hashWithSalt` (r ^. (robotEntity . entityHash)) `h
 robotKnows :: Robot -> Entity -> Bool
 robotKnows r e = contains0plus e (r ^. robotInventory) || contains0plus e (r ^. equippedDevices)
 
+isInteractive :: Robot -> Bool
+isInteractive r = not $ r ^. robotDisplay . invisible && r ^. systemRobot
```

Comment:
We may want to improve this heuristic, or make it an explicit "interactive" property.

For example, I don't think we ever want to `meet` a "growing" `flower`.

Reply:
Yes, I agree, growing `tree`s, `flower`s, etc. are going to be a problem here.  We use robots internally to simulate growth --- but from the perspective of a player that is just an implementation detail, they should not be returned by `meet`.  This heuristic is better than nothing, but we will definitely want to improve it.  I am fine with either merging this as is and making an issue to improve it later, or coming up with a better heuristic in this PR.

An explicit boolean `interactive` property probably makes sense, and I think it makes sense for `interactive = true` to be the default, but invisible system robots can still be non-interactive by default.